docs/template_plugin/src/template_pattern_transformation.hpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <vector>
+#include <memory>
+
+#include <ngraph/pass/graph_rewrite.hpp>
+
+// ! [graph_rewrite:template_transformation_hpp]
+// template_pattern_transformation.hpp
+class MyPatternBasedTransformation: public ngraph::pass::GraphRewrite {
+public:
+    MyPatternBasedTransformation() : GraphRewrite() {
```

Comment:
we need to insert call of these transformations in https://github.com/openvinotoolkit/openvino/blob/master/docs/template_plugin/src/template_executable_network.cpp#L97 for demonstration purposes

Reply:
and refer to this guide from https://github.com/openvinotoolkit/openvino/blob/master/docs/IE_PLUGIN_DG/ExecutableNetwork.md

Reply:
I added call to template_executable_network.cpp. Link to transformations guide already exists in ExecutableNetwork.md